cron/monthly/data-export.js

```diff
@@ -143,16 +145,136 @@ async function run() {
       console.log(err);
     }
   });
-  console.log('done');
-  process.exit(0);
+  console.log('all files created');
 }
 
-run();
+const driveCredentials = {
+  clientId: process.env.DRIVE_CLIENT_ID,
+  clientSecret: process.env.DRIVE_CLIENT_SECRET,
+  redirectUri: process.env.DRIVE_REDIRECT_URI,
+};
+
+// If modifying these scopes, delete token.json.
+const SCOPES = ['https://www.googleapis.com/auth/drive'];
+// The file token.json stores the user's access and refresh tokens, and is
+// created automatically when the authorization flow completes for the first
+// time.
+const TOKEN_PATH = 'token.json';
+
+/**
+ * Create an OAuth2 client with the given credentials, and then execute the
+ * given callback function.
+ * @param {Object} credentials The authorization client credentials.
+ * @param {function} callback The callback to call with the authorized client.
+ */
+const authorize = (driveCredentials, callback) => {
+  const { clientSecret, clientId, redirectUri } = driveCredentials;
+  const oAuth2Client = new google.auth.OAuth2(clientId, clientSecret, redirectUri);
+
+  // Check if we have previously stored a token.
+  fs.readFile(TOKEN_PATH, (err, token) => {
+    if (err) return getAccessToken(oAuth2Client, callback);
+    oAuth2Client.setCredentials(JSON.parse(token));
+    callback(oAuth2Client);
+  });
+};
+
+/**
+ * Get and store new token after prompting for user authorization, and then
+ * execute the given callback with the authorized OAuth2 client.
+ * @param {google.auth.OAuth2} oAuth2Client The OAuth2 client to get token for.
+ * @param {getEventsCallback} callback The callback for the authorized client.
+ */
+const getAccessToken = (oAuth2Client, callback) => {
+  const authUrl = oAuth2Client.generateAuthUrl({
+    access_type: 'offline', // eslint-disable-line camelcase
+    scope: SCOPES,
+  });
+  console.log('Authorize this app by visiting this url:', authUrl);
+  const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+  });
+  rl.question('Enter the code from that page here: ', code => {
+    rl.close();
+    oAuth2Client.getToken(code, (err, token) => {
+      if (err) return console.error('Error retrieving access token', err);
+      oAuth2Client.setCredentials(token);
+      // Store the token to disk for later program executions
+      fs.writeFile(TOKEN_PATH, JSON.stringify(token), err => {
+        if (err) return console.error(err);
+        console.log('Token stored to', TOKEN_PATH);
+      });
+      callback(oAuth2Client);
+    });
+  });
+};
+
+/**
+ * upload file to google drive
+ * @param {google.auth.OAuth2} auth An authorized OAuth2 client.
+ */
+const uploadFiles = async auth => {
+  const drive = google.drive({ version: 'v3', auth });
+
+  //create a folder on drive
+  const folderMetadata = {
+    name: `oc-data-${startDate.getFullYear()}-${month}`,
+    mimeType: 'application/vnd.google-apps.folder',
+  };
+  const folder = await drive.files.create({
```

Comment:
No. It will create another folder with the same name.
![Screenshot at 2020-08-31 16-18-23](https://user-images.githubusercontent.com/46647141/91712607-b8c0dd80-eba5-11ea-8f55-52523a3efb50.png)